test/e2e-api/admin/posts.test.js

```diff
@@ -490,6 +494,131 @@ describe('Posts API', function () {
         should(model.get('newsletter_id')).eql(newsletterId);
     });
 
+    it('Defaults to the default newsletter when publishing without a newsletter_id', async function () {
+        const post = {
+            title: 'My post without newsletter_id',
+            status: 'draft',
+            feature_image_alt: 'Testing newsletter_id',
+            feature_image_caption: 'Testing <b>feature image caption</b>',
+            mobiledoc: testUtils.DataGenerator.markdownToMobiledoc('my post'),
+            created_at: moment().subtract(2, 'days').toDate(),
+            updated_at: moment().subtract(2, 'days').toDate(),
+            created_by: ObjectId().toHexString(),
+            updated_by: ObjectId().toHexString()
+        };
+
+        const res = await request.post(localUtils.API.getApiQuery('posts'))
+            .set('Origin', config.get('url'))
+            .send({posts: [post]})
+            .expect('Content-Type', /json/)
+            .expect('Cache-Control', testUtils.cacheRules.private)
+            .expect(201);
+
+        const id = res.body.posts[0].id;
+
+        const updatedPost = {
+            status: 'published',
+            updated_at: res.body.posts[0].updated_at
+        };
+
+        await request
+            .put(localUtils.API.getApiQuery('posts/' + id + '/?email_recipient_filter=all'))
+            .set('Origin', config.get('url'))
+            .send({posts: [updatedPost]})
+            .expect('Content-Type', /json/)
+            .expect('Cache-Control', testUtils.cacheRules.private)
+            .expect(200);
+
+        const model = await models.Post.findOne({
+            id
+        }, testUtils.context.internal);
+
+        should(model.get('newsletter_id')).not.eql(null);
```

Comment:
This is only testing that newsletter_id is set, not that it's set to the default newsletter which is what the test claims